llmfoundry/data/utils.py

```diff
@@ -20,6 +20,55 @@
 log = logging.getLogger(__name__)
 
 
+class LossGeneratingTokensCollatorWrapper:
+    """Collator wrapper to add loss generating token counts to batch."""
+
+    def __init__(
+        self,
+        base_collator: Callable,
+        token_counting_func: Callable[[Batch], Union[int, dict[str, int]]],
```

Comment:
Do we need to pass the counting fn as an arg? given there's only one option and no way to pipe it through? Why not just call fn directly